python/ray/serve/scripts.py

```diff
@@ -232,49 +234,170 @@ def start(
+@click.option(
+    "--runtime-env",
+    type=str,
+    default=None,
+    required=False,
+    help="Path to a local YAML file containing a runtime_env definition.",
+)
+@click.option(
+    "--runtime-env-json",
+    type=str,
+    default=None,
+    required=False,
+    help="JSON-serialized runtime_env dictionary.",
+)
+@click.option(
+    "--working-dir",
+    type=str,
+    default=None,
+    required=False,
+    help=(
+        "Directory containing files that your application(s) will run in. When using "
+        "the 'local' provider, this must be a remote URI to a .zip file (e.g., S3 "
+        "bucket). Other providers may support local paths. This overrides the "
+        "working_dir in --runtime-env if both are specified."
+    ),
+)
+@click.option(
+    "--name",
+    required=False,
+    default=None,
+    type=str,
+    help="Custom name for the application(s).",
+)
+@click.option(
+    "--base-image",
+    required=False,
+    default=None,
+    type=str,
+    help=(
+        "Container image to use for the application(s). Not supported for the "
+        "default 'local' provider because it deploys to an existing cluster."
+    ),
+)
+@click.option(
+    "--provider",
+    required=False,
+    default=None,
+    type=str,
+    help=(
+        "Deploy provider to use. By default, this uses a 'local' provider that makes "
+        "a REST API request to the provided address. Can also be set with the "
+        f"{DEPLOY_PROVIDER_ENV_VAR} environment variable."
     ),
 )
-@click.argument("config_file_name")
 @click.option(
     "--address",
     "-a",
     default=os.environ.get("RAY_DASHBOARD_ADDRESS", "http://localhost:8265"),
     required=False,
     type=str,
-    help=RAY_DASHBOARD_ADDRESS_HELP_STR,
+    help=RAY_DASHBOARD_ADDRESS_HELP_STR + " . Only used by the 'local' provider.",
 )
-def deploy(config_file_name: str, address: str):
-    warn_if_agent_address_set()
-
-    with open(config_file_name, "r") as config_file:
-        config = yaml.safe_load(config_file)
+def deploy(
+    config_or_import_path: str,
+    arguments: Tuple[str],
+    runtime_env: str,
+    runtime_env_json: str,
+    working_dir: str,
+    name: Optional[str],
+    base_image: Optional[str],
+    provider: Optional[str],
+    address: str,
+):
+    args_dict = convert_args_to_dict(arguments)
+    final_runtime_env = parse_runtime_env_args(
+        runtime_env=runtime_env,
+        runtime_env_json=runtime_env_json,
+        working_dir=working_dir,
+    )
 
-    ServeDeploySchema.parse_obj(config)
-    ServeSubmissionClient(address).deploy_applications(config)
+    # Skip validating runtime_env URIs so publishers can enable local URI usage.
+    with _skip_validating_runtime_env_uris():
+        config = _generate_config_from_file_or_import_path(
+            config_or_import_path,
+            name=name,
+            arguments=args_dict,
+            runtime_env=final_runtime_env,
+        )
```

Comment:
Could we skip validation only if the user isn't using the `local` provider? If they are using the `local` provider, then they must provide a remote URI (according to [this comment](https://github.com/ray-project/ray/pull/42889/files#diff-309b33091c76269e3d557d272f26e8cb5a9cda6fb4e0917bc05dc65d2c780ddbR315-R320)), so we should validate their URIs.

Reply:
it'll get validated inside the local provider

Reply:
I have a better idea for this... let me add a method to the provider "supports_local_uris" and then selectively disable it

Reply:
updated, this is way better!